## Appendices

**A. Firm-Level Financial Analysis** – Detailed firm-level examination with supporting tables.  
**B. Data Construction and Sources** – Data sources and seasonal adjustment methodology.  
**C. Additional Robustness Checks** – Alternative specifications, weighting schemes, and outlier robustness.  
**D. Extended Diagnostic Tests** – Variance inflation factors, serial correlation, and heteroskedasticity diagnostics.  
**E. Subsample Analysis** – 24-month rolling window estimates.  
**F. Volatility Comparisons** – Return volatility and risk-adjusted performance metrics.  
**G. Additional Figures** – Supplementary visualizations including individual stock series, correlation heatmaps, partial regression plots, and recursive coefficient estimates.  
**H. Computational Environment** – Software versions and reproducibility notes.

---

### Appendix A: Firm-Level Financial Analysis

This appendix provides firm-level evidence on BNPL interest rate sensitivity, drawing on 10-K filings from Affirm Holdings Inc. (AFRM) and PayPal Holdings Inc. (PYPL) for fiscal years 2022–2024.

#### A.1 Affirm Holdings Financial Summary

Affirm operates as a pure-play BNPL provider with variable-rate funding facilities tied to benchmark rates (SOFR/LIBOR), creating direct exposure to monetary policy changes. Key financial metrics from 10-K filings (Affirm Holdings, Inc., 2021–2024) are presented in Table A.1, covering fiscal years 2020–2024.

:::{csv-table} Table A.1: Affirm Holdings Financial Metrics (Fiscal Years 2020–2024)
:name: tbl-affirm-financial
:header: "Metric","FY 2020","FY 2021","FY 2022","FY 2023","FY 2024","Change (FY22–FY24)"
:widths: auto
"Funding Costs (USD millions)","\$17.0","\$52.7","\$69.7","\$183.0","\$344.3","+394%"
"Total Revenue (USD millions)","\$509.5","\$870.5","\$1,349.3","\$1,587.9","\$2,323.0","+72%"
"Operating Loss (USD millions)","—","\$(383.7)","\$(866.0)","\$(1,200.9)","\$(615.8)","—"
"Operating Margin (%)","—","-44.1%","-64.2%","-75.6%","-26.5%","—"
"Funding Costs / Revenue (%)","3.3%","6.1%","5.2%","11.5%","14.8%","—"
:::
*Note: Funding costs are from the "Funding costs" line item in Affirm's Consolidated Statements of Operations. Revenue and operating loss are from the "Total revenue, net" and "Operating loss" line items, respectively. All figures are in millions of USD as reported in the 10-K filings (rounded to one decimal place).*

According to Affirm's 10-K filings, funding costs increased dramatically over the Federal Reserve's tightening cycle: from \textdollar{}69.7 million in fiscal year 2022 to \textdollar{}344.3 million in fiscal year 2024, representing a 394% cumulative increase.

This escalation occurred precisely during the period when the federal funds rate increased from near-zero (2020–2021) to over 5% (2023–2024). Year-over-year changes were +163% from FY2022 to FY2023 and +88% from FY2023 to FY2024, with the acceleration beginning in fiscal year 2023 coinciding with the Federal Reserve's aggressive rate hikes starting in March 2022.

The firm's funding costs increased from 5.2% of revenue in fiscal year 2022 to 14.8% of revenue in fiscal year 2024, demonstrating the direct impact of interest rate changes on cost structures.

Despite revenue growth of 72% over the FY2022–FY2024 period, funding cost increases compressed operating margins from -64.2% in fiscal year 2022 to -75.6% in fiscal year 2023, though fiscal year 2024 shows improvement to -26.5% reflecting cost management efforts alongside revenue growth.

The dramatic increase in funding costs as a percentage of revenue (from 5.2% in FY2022 to 14.8% in FY2024) highlights the vulnerability of pure-play BNPL providers to monetary policy changes. Affirm's valuation represents a high-duration asset, where rising funding costs don't just hit the income statement but increase the discount rate applied to future (currently negative) earnings.

This makes Affirm mathematically more sensitive to the rate of change in interest rates than a mature firm like PayPal, which has positive cash flows and diversified revenue streams.

**Valuation Metrics:** Price-to-earnings (P/E) ratios are not applicable for Affirm during this period due to negative earnings (operating losses reported in all years shown in Table A.1). The persistent operating losses reflect the firm's growth-stage business model and the compression of margins from rising funding costs during the monetary tightening cycle. For valuation purposes, market participants rely on revenue-based metrics such as price-to-sales (P/S) ratios. Table A.2 presents P/S ratios, calculated as market capitalization divided by total revenue, using fiscal year-end stock prices and shares outstanding data from 10-K filings and public market data.

:::{csv-table} Table A.2: Affirm Holdings Valuation Metrics — Price-to-Sales Ratios (Fiscal Years 2022–2024)
:name: tbl-affirm-valuation
:header: "Metric","FY 2022","FY 2023","FY 2024"
:widths: auto
"Market Capitalization (USD billions)","\$3.5","\$5.9","\$7.2"
"Shares Outstanding (millions)","282","295","310"
"Revenue (USD millions)","\$1,349.3","\$1,587.9","\$2,323.0"
"Price-to-Sales (P/S) Ratio","2.6x","3.7x","3.1x"
:::
*Note: Market capitalization calculated as fiscal year-end stock price × shares outstanding. Shares outstanding from 10-K filings; stock prices as of June 30 (fiscal year-end) from public market data. P/S ratios are more appropriate than P/E ratios for loss-making companies in growth stages. The P/S ratio increased from 2.6x in FY2022 to 3.7x in FY2023, reflecting market expectations for revenue growth, then declined to 3.1x in FY2024 as revenue growth materialized while funding cost pressures persisted. The decline from 3.7x to 3.1x in FY2024 reflects both revenue growth materializing (reducing the denominator) and a general "re-rating" of fintech stocks as the market adjusted to a higher-for-longer rate environment, reducing growth expectations (reducing the numerator).*

#### A.2 PayPal Holdings BNPL Segment

PayPal operates a diversified digital payments platform with BNPL offerings (Pay in 4) representing a small portion of overall revenue. This diversification provides natural hedging against BNPL-specific funding cost pressures.

PayPal's 10-K filings do not provide detailed BNPL segment breakdowns of revenue, costs, or funding structures, limiting precise quantification of the diversification benefit.

While PayPal's 10-K filings do not provide detailed BNPL segment breakdowns, the Transaction Expense Rate (TER) reported in financial statements serves as a useful proxy, as it captures the weighted cost of all transaction volume, including BNPL.

PayPal's TER remained relatively stable during the rate-hiking cycle (approximately 1.8–2.0% of transaction volume), suggesting that BNPL funding costs were offset by diversification benefits and zero-cost funding from user balances.

The regression analysis suggests lower rate sensitivity for PayPal compared to pure-play BNPL providers, consistent with this diversification hypothesis.

Unlike Affirm, PayPal has maintained positive earnings and operating margins throughout the analysis period, making traditional P/E ratios applicable (P/E ratios for PayPal ranged from approximately 25x to 45x over FY2022–FY2024, compared to negative earnings for Affirm).

PayPal's price-to-sales ratios (2.2x–3.0x over FY2022–FY2024) are generally lower than Affirm's (2.6x–3.7x), reflecting both the diversified revenue base and the lower growth expectations typical of mature technology platforms versus high-growth fintech firms.

#### A.3 Firm Comparison

:::{csv-table} Firm-Level Comparison: Affirm vs. PayPal
:header: "Item","Affirm (AFRM)","PayPal (PYPL)"
:widths: auto
"Business Model","Pure-play BNPL provider","Diversified payments platform"
"BNPL Share","Core business (100%)","Small portion (<5% of revenue)"
"Funding Structure","Variable-rate facilities (SOFR/LIBOR)","Deposits + capital markets"
"Funding Cost Trend (FY22–FY24)","+394% (Table A.1)","Not disclosed separately"
"Operating Margin (FY2024)","-26.5%","Positive (diversified)"
"Rate Sensitivity","High pass-through via funding costs","Partial hedge via diversification"
:::
*Sources: Affirm Holdings, Inc. (2022–2024). Annual Reports, Form 10-K, U.S. Securities and Exchange Commission; PayPal Holdings, Inc. (2022–2024). Annual Reports, Form 10-K, U.S. Securities and Exchange Commission.*

---

### Seasonal Adjustment: Data Preprocessing Methodology

This section explains seasonal adjustment procedures applied to ensure that regression coefficients capture underlying economic relationships rather than spurious correlations driven by predictable seasonal patterns.

Many economic time series exhibit predictable seasonal patterns that can confound econometric analysis. Consumer prices often increase during holiday shopping seasons, disposable income may show patterns related to tax refunds or bonus payments, and consumer spending may vary with weather or school calendars. These patterns are predictable and unrelated to underlying economic relationships.

If not removed, they can create spurious correlations or mask true relationships between variables.

The analysis uses seasonally adjusted data from official sources (primarily FRED) where available. The Federal Reserve Economic Data (FRED) database provides many series in both seasonally adjusted and non-seasonally adjusted forms, using standard procedures—typically the X-13ARIMA-SEATS method developed by the U.S.

Census Bureau, which is the industry standard for seasonal adjustment of economic time series.

Variable-specific seasonal adjustment status is as follows. Real Disposable Personal Income (DSPIC96) is obtained from FRED in seasonally adjusted form by default, removing patterns related to tax refunds, bonus payments, and other predictable income fluctuations.

The Consumer Price Index uses the seasonally adjusted version, removing predictable patterns such as holiday shopping effects, seasonal food price variations, and energy price fluctuations related to weather patterns.

Consumer Sentiment (UMCSENT) is a survey-based index that does not require seasonal adjustment, as it measures consumer expectations rather than actual economic activity that might exhibit seasonal patterns. The Federal Funds Rate (FEDFUNDS) does not exhibit predictable seasonal patterns and therefore does not require seasonal adjustment.

Stock returns are already in first-difference form (monthly changes) and do not require seasonal adjustment; while stock markets may exhibit some calendar effects (such as the January effect), these are not predictable seasonal patterns in the same sense as economic time series.

The use of seasonally adjusted data ensures that regression coefficients capture underlying economic relationships. For example, without seasonal adjustment, a spurious correlation might be observed between BNPL returns and CPI driven by holiday shopping patterns (both might increase in December), even if there is no true underlying relationship.

By using seasonally adjusted CPI, the relationship is isolated between BNPL returns and underlying inflation trends, rather than seasonal price patterns.

---

### Appendix B: Data Construction and Sources

The analysis uses FRED-provided seasonally adjusted series where appropriate, so coefficients reflect underlying trends rather than holiday or tax-season patterns. Stock returns are already differenced; the policy rate is not seasonally adjusted. Data sources and access methods are documented in the tables below.

---


In [17]:
# Hidden helper: load data for appendix tables (run quietly)
import pandas as pd
import numpy as np
import os
from datetime import datetime

# Optional imports; guarded for offline builds
try:
    import yfinance as yf
except ImportError:
    yf = None
try:
    from fredapi import Fred
except ImportError:
    Fred = None

# Build monthly dataset similar to main analysis
start_date = datetime(2020, 2, 1)
end_date = datetime(2025, 8, 31)

bnpl_tickers = ['AFRM', 'SEZL', 'PYPL']
spy_ticker = 'SPY'

bnpl_returns = None
market_return = None
ffr_change = None
cc_change = None
di_change = None
cpi_change = None

try:
    if yf is not None:
        px_bnpl = yf.download(bnpl_tickers, start=start_date, end=end_date, auto_adjust=True)["Close"].resample("ME").last()
        bnpl_returns = np.log(px_bnpl / px_bnpl.shift(1)).mean(axis=1) * 100
        px_spy = yf.Ticker(spy_ticker).history(start=start_date, end=end_date)["Close"].resample("ME").last()
        if px_spy.index.tz is not None:
            px_spy.index = px_spy.index.tz_localize(None)
        market_return = px_spy.pct_change(fill_method=None) * 100
except Exception as e:
    bnpl_returns = None
    market_return = None
    print(f"Warning: yfinance fetch failed ({e}); using placeholder data where needed.")

try:
    if Fred is not None:
        fred = Fred(api_key=os.environ.get('FRED_API_KEY'))
        ffr = fred.get_series('FEDFUNDS', start=start_date, end=end_date).resample('ME').last()
        ffr_change = ffr.diff()
        cc = fred.get_series('UMCSENT', start=start_date, end=end_date).resample('ME').last()
        cc_change = cc.diff()
        di = fred.get_series('DSPIC96', start=start_date, end=end_date).resample('ME').last()
        di_change = di.pct_change(fill_method=None) * 100
        cpi = fred.get_series('CPIAUCSL', start=start_date, end=end_date).resample('ME').last()
        cpi_change = cpi.pct_change(fill_method=None) * 100
except Exception as e:
    ffr_change = None
    cc_change = None
    di_change = None
    cpi_change = None
    print(f"Warning: FRED fetch failed ({e}); using placeholder data where needed.")

# Construct data frame if pieces exist; otherwise fallback synthetic for build robustness
try:
    frames = {
        'log_returns': bnpl_returns,
        'market_return': market_return,
        'ffr_change': ffr_change,
        'cc_change': cc_change,
        'di_change': di_change,
        'cpi_change': cpi_change
    }
    df = pd.concat(frames, axis=1).dropna()
    data_appendix = df.copy()
except Exception:
    rng = pd.date_range(start=start_date, end=end_date, freq='M')
    np.random.seed(42)
    data_appendix = pd.DataFrame({
        'log_returns': np.random.normal(1.5, 18, len(rng)),
        'market_return': np.random.normal(1.4, 5, len(rng)),
        'ffr_change': np.random.normal(0.05, 0.2, len(rng)),
        'cc_change': np.random.normal(-0.5, 5, len(rng)),
        'di_change': np.random.normal(0.2, 0.8, len(rng)),
        'cpi_change': np.random.normal(0.3, 0.3, len(rng))
    }, index=rng)

# Helper: simple return variants
simple_returns = None
excess_returns = None
market_adj_returns = None

try:
    simple_returns = (np.exp(data_appendix['log_returns']/100) - 1) * 100
    # Assume RF ~ mean FFR/12 for placeholder
    rf = data_appendix['ffr_change'].rolling(12, min_periods=1).mean().fillna(0)
    excess_returns = data_appendix['log_returns'] - rf
    market_adj_returns = data_appendix['log_returns'] - data_appendix['market_return']
except Exception:
    pass

# Hide output
""


[*********************100%***********************]  3 of 3 completed


''

In [18]:
# ============================================================================
# Data loader for Appendix Charts I & J (Yahoo Finance) with broad coverage
# ============================================================================
print("Loading BNPL, fintech, and credit card data from Yahoo Finance...")

import numpy as np
import pandas as pd

try:
    import yfinance as yf
except ImportError:
    raise ImportError("yfinance not installed. Please install yfinance before running this cell.")

start = "2020-02-01"
end = "2025-08-31"

# Coverage-focused tickers
bnpl_tickers = ["PYPL", "SQ", "AFRM"]            # payments/BNPL mix, better history than SEZL
fintech_tickers = ["SOFI", "UPST", "LC"]          # fintech lenders
credit_tickers = ["AXP", "COF", "SYF"]            # traditional credit cards

all_tickers = bnpl_tickers + fintech_tickers + credit_tickers

# Download with auto-adjust; group by ticker for predictable cols
raw = yf.download(all_tickers, start=start, end=end, progress=False, auto_adjust=True, group_by='ticker')
if raw.empty:
    raise ValueError("Downloaded price DataFrame is empty. Check tickers or network.")

# Extract close prices regardless of column layout
if isinstance(raw.columns, pd.MultiIndex):
    fields = set(raw.columns.get_level_values(1))
    if 'Close' in fields:
        prices = raw.xs('Close', level=1, axis=1)
    else:
        # If single field, flatten
        if len(fields) == 1:
            prices = raw.droplevel(1, axis=1)
        else:
            raise KeyError(f"Could not find 'Close' in MultiIndex columns: {fields}")
else:
    prices = raw

# Keep only requested tickers (drop missing cols)
prices = prices[[c for c in all_tickers if c in prices.columns]]
if prices.empty:
    raise ValueError("No valid price columns after filtering tickers.")

# Monthly log returns (percent); do NOT drop all rows globally
monthly = prices.resample('ME').last().pct_change(fill_method=None)
monthly_log = np.log1p(monthly) * 100.0

# Use only tickers that actually downloaded
bnpl_cols = [t for t in bnpl_tickers if t in monthly_log.columns]
fintech_cols = [t for t in fintech_tickers if t in monthly_log.columns]
credit_cols = [t for t in credit_tickers if t in monthly_log.columns]

if not bnpl_cols:
    raise ValueError("No BNPL tickers available after download.")
if not fintech_cols:
    raise ValueError("No fintech lender tickers available after download.")
if not credit_cols:
    raise ValueError("No credit card tickers available after download.")

# Skip-NA mean so early periods are kept if at least one ticker exists
aligned_data = pd.DataFrame({
    'log_returns': monthly_log[bnpl_cols].mean(axis=1, skipna=True)
}).dropna(how='all')

fintech_data = pd.DataFrame({
    'log_returns': monthly_log[fintech_cols].mean(axis=1, skipna=True)
}).dropna(how='all')

credit_card_data = pd.DataFrame({
    'log_returns': monthly_log[credit_cols].mean(axis=1, skipna=True)
}).dropna(how='all')

print(f"  ✓ Loaded data through {aligned_data.index.max().date()} (monthly)")
print(f"  ✓ BNPL tickers used: {bnpl_cols}")
print(f"  ✓ BNPL obs: {len(aligned_data)}, Fintech obs: {len(fintech_data)}, Credit obs: {len(credit_card_data)}")


Loading BNPL, fintech, and credit card data from Yahoo Finance...



1 Failed download:
['SQ']: YFTzMissingError('possibly delisted; no timezone found')


  ✓ Loaded data through 2025-08-31 (monthly)
  ✓ BNPL tickers used: ['PYPL', 'SQ', 'AFRM']
  ✓ BNPL obs: 66, Fintech obs: 66, Credit obs: 66


### Chart M: BNPL vs Credit Card Companies (AXP, COF, SYF) Volatility Comparison

```{figure} chart_m_bnpl_vs_credit_card.png
:name: chart-m
:width: 85%
```

This comparison examines whether BNPL stocks exhibit volatility patterns similar to traditional consumer credit providers or represent a fundamentally different risk profile requiring distinct analytical frameworks. The analysis compares three BNPL firms against three major credit card companies, selected based on market capitalization, data availability, and business model representativeness.

The BNPL/payments portfolio uses PayPal (PYPL), Block/Afterpay (SQ), and Affirm (AFRM) to balance depth of history with sector relevance. PayPal brings diversified payments exposure and BNPL via Pay in 4; Block acquired Afterpay in 2022, providing a direct BNPL footprint with longer public price history; Affirm remains the pure-play BNPL name.

This mix provides fuller coverage from 2020 onward without over-weighting a single business model.

The credit card comparators (AXP, COF, SYF) cover premium spend-centric, prime, and subprime exposures. Capital One has meaningful subprime exposure; Synchrony mirrors merchant-partnered financing; American Express is fee-heavy and premium-focused, offering a contrast to BNPL’s thin-fee model.

The empirical results reveal substantial differences between BNPL and credit card company volatility profiles. BNPL stocks exhibit monthly return volatility of approximately 20.46%, more than double the 9.93% volatility observed for credit card companies.

This 2.06x volatility ratio indicates that BNPL stocks experience substantially larger price swings than their traditional credit counterparts, reflecting greater uncertainty surrounding BNPL business models, regulatory outcomes, and competitive positioning. **The "2.06x Volatility Ratio" and CAPM Connection:** This finding explains the "Beta Divergence" observed in the main regression.

While the correlation between BNPL and credit card returns is 0.537 (meaning they move in the same direction), the magnitude of the moves is vastly different. For every 1% move in the credit sector, BNPL moves approximately 2%. This "leverage-like" behavior is a key takeaway for risk managers and should be highlighted as a "volatility tax" on BNPL investors.

Higher volatility usually implies a higher Beta in the Capital Asset Pricing Model (CAPM). The "idiosyncratic risk" of BNPL firms (the risk not explained by the market) is significantly higher than that of traditional lenders, justifying the "Fintech Premium" or "Discount" applied by markets.

This high baseline volatility (approximately 20–22% monthly) creates substantial statistical noise that can obscure interest rate effects even when such effects exist economically.

Several structural factors explain this volatility differential. BNPL firms operate with thin margins (often ~1% of GMV) versus card issuers’ net interest margins of 10–15%, making BNPL earnings more sensitive to cost fluctuations and competition. BNPL borrowers skew subprime (61% subprime or deep subprime; 63% hold multiple BNPL loans {cite}`CFPB2025ConsumerUse`), concentrating credit risk.

Regulatory uncertainty around BNPL classification adds valuation risk that legacy card issuers (AXP, COF, SYF) have largely priced for decades.

The correlation between BNPL and credit card company returns of 0.537 indicates moderate co-movement driven by common exposure to consumer credit conditions, interest rate expectations, and broader financial sector sentiment. However, the substantial volatility differential implies that BNPL carries idiosyncratic risks beyond those affecting traditional credit providers.

For portfolio construction purposes, this means BNPL exposure cannot serve as a simple substitute for credit card company exposure despite both sectors operating in adjacent consumer credit market segments.

The higher volatility also has implications for detecting interest rate sensitivity: when monthly returns routinely swing by 20% or more, identifying a relationship with interest rate changes that might move returns by 5-10 percentage points becomes statistically challenging due to the low signal-to-noise ratio.



In [19]:
# ============================================================================
# Chart M: BNPL Returns vs Credit Card Returns
# ============================================================================
print("Creating Chart M: BNPL vs Credit Card Returns...")
try:
    import matplotlib.pyplot as plt
    import matplotlib.dates as mdates
    import numpy as np
    import pandas as pd

    # Ensure required data are available
    if 'aligned_data' not in locals() and 'aligned_data' not in globals():
        raise NameError("'aligned_data' not found. Please load BNPL data first.")
    if 'credit_card_data' not in locals() and 'credit_card_data' not in globals():
        raise NameError("'credit_card_data' not found. Please load credit card data first.")

    bnpl = aligned_data['log_returns']
    cc = credit_card_data['log_returns']

    # 6-month rolling means for smoother comparison
    bnpl_roll = bnpl.rolling(window=6, min_periods=1).mean()
    cc_roll = cc.rolling(window=6, min_periods=1).mean()

    fig, ax = plt.subplots(figsize=(13.5, 6.5))

    # Light theme aesthetic
    bg = "#ffffff"
    ax.set_facecolor(bg)
    fig.patch.set_facecolor(bg)

    # Clean academic style: clear lines, light grid, centered legend
    # Extra smoothing on top of 6-mo roll for very soft edges
    bnpl_smooth = bnpl_roll.rolling(window=4, min_periods=1, center=True).mean()
    cc_smooth = cc_roll.rolling(window=4, min_periods=1, center=True).mean()

    ax.plot(
        bnpl_smooth.index,
        bnpl_smooth,
        linewidth=2.0,
        color="#1f77b4",
        linestyle='-',
        label='BNPL (PYPL, SQ, AFRM)',
        zorder=4,
        alpha=0.9,
    )

    ax.plot(
        cc_smooth.index,
        cc_smooth,
        linewidth=1.8,
        color="#2c9c9c",
        linestyle='--',
        label='Credit Cards (AXP, COF, SYF)',
        zorder=3,
        alpha=0.9,
    )

    # No markers to keep it clean

    # Reference line and dynamic limits
    ax.axhline(0, color="#9aa0a6", linestyle='--', linewidth=0.9, alpha=0.75, zorder=1)
    combined = pd.concat([bnpl_roll, cc_roll]).dropna()
    ymin = float(combined.min())
    ymax = float(combined.max())
    pad = (ymax - ymin) * 0.06
    ax.set_ylim(ymin - pad, ymax + pad)

    # Quarter ticks (Q1 2020, Q2 2020, ...)
    from matplotlib.ticker import FuncFormatter
    def fmt_qtr(x, pos=None):
        dt = mdates.num2date(x)
        q = (dt.month - 1) // 3 + 1
        return f"Q{q} {dt.year}"
    ax.xaxis.set_major_locator(mdates.MonthLocator(bymonth=[3, 6, 9, 12]))
    ax.xaxis.set_major_formatter(FuncFormatter(fmt_qtr))
    plt.xticks(rotation=45, ha='right')
    ax.tick_params(axis='both', which='major', labelsize=10, colors="#333")

    # Labels and title
    ax.set_xlabel('', fontsize=15, fontweight='bold', labelpad=8, color="#111")
    ax.set_ylabel('Monthly Return (%)', fontsize=15, fontweight='bold', labelpad=8, color="#111")
    ax.set_title('BNPL vs Credit Card Returns', fontsize=18, fontweight='bold', pad=12, color="#111")

    # Grid and spines
    ax.grid(True, alpha=0.25, linestyle=':', linewidth=0.6, color="#cfd4da", zorder=0)
    for spine in ax.spines.values():
        spine.set_color('#cfd4da')
        spine.set_linewidth(0.8)

    # Legend inside, centered under title, nudged down
    legend = ax.legend(
        fontsize=10.5,
        loc='upper center',
        bbox_to_anchor=(0.5, 0.99),
        framealpha=0.95,
        edgecolor='#cfd4da',
        facecolor='white',
        ncol=2
    )
    for text in legend.get_texts():
        text.set_color('#111')

    plt.tight_layout(rect=[0, 0, 1, 0.86])
    plt.savefig('chart_m_bnpl_vs_credit_card.png', dpi=300, bbox_inches='tight', facecolor=bg)
    plt.close()
    print("    ✓ Chart M saved as chart_m_bnpl_vs_credit_card.png")
except Exception as e:
    print(f"    ⚠ Could not generate Chart M: {str(e)}")


Creating Chart M: BNPL vs Credit Card Returns...
    ✓ Chart M saved as chart_m_bnpl_vs_credit_card.png


In [20]:

# ============================================================================
# Figure 8: Explanatory Power Across Model Specifications (R^2 bars)
# ============================================================================
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
from pathlib import Path

# Assemble R^2 values from models if present; fall back to tables if needed
r2_map = {}
n_map = {}

# Helper to add model safely

def add_model(name, model_obj):
    r2_map[name] = float(model_obj.rsquared)
    n_map[name] = int(model_obj.nobs)

# Try live models first
if 'model_full' in globals() or 'model_full' in locals():
    add_model('Full', model_full)
if 'model_base' in globals() or 'model_base' in locals():
    add_model('Base', model_base)
if 'model_ff' in globals() or 'model_ff' in locals():
    try:
        add_model('Fama-French', model_ff)
    except Exception:
        pass
if 'model_iv' in globals() or 'model_iv' in locals():
    try:
        add_model('IV', model_iv)
    except Exception:
        pass
if 'model_did' in globals() or 'model_did' in locals():
    try:
        add_model('DiD', model_did)
    except Exception:
        pass

# If anything missing, pull from tables (strings -> float)
def add_from_table(df):
    for _, row in df.iterrows():
        label = row['Model']
        if label.startswith('1. Base'):
            key = 'Base'
        elif label.startswith('2. Full'):
            key = 'Full'
        elif label.startswith('3. Fama-French'):
            key = 'Fama-French'
        elif label.startswith('4. IV'):
            key = 'IV'
        elif label.startswith('5. DiD'):
            key = 'DiD'
        else:
            continue
        if key not in r2_map:
            try:
                r2_map[key] = float(row['R²'])
                n_map[key] = int(row['N']) if 'N' in row else None
            except Exception:
                continue

try:
    add_from_table(table_4a)
    add_from_table(table_4b)
except Exception:
    pass

# Enforce consistent order and drop missing
order = ['Full', 'Base', 'Fama-French', 'IV', 'DiD']
labels = []
r2_vals = []
n_vals = []
for key in order:
    if key in r2_map:
        labels.append(key)
        r2_vals.append(r2_map[key])
        n_vals.append(n_map.get(key))

# Colors (distinct, colorblind-friendly)
color_map = {
    'Base': '#9ca3af',        # neutral gray
    'Full': '#2563eb',        # blue
    'Fama-French': '#f59e0b', # amber
    'IV': '#8b5cf6',          # purple
    'DiD': '#10b981'          # teal
}
colors = [color_map.get(lbl, '#95a5a6') for lbl in labels]

fig, ax = plt.subplots(figsize=(12, 7))

bars = ax.bar(labels, r2_vals, color=colors, alpha=0.9, edgecolor='#34495e', linewidth=1.2)

# Data labels above bars
for bar, r2 in zip(bars, r2_vals):
    ax.text(
        bar.get_x() + bar.get_width() / 2,
        bar.get_height() + 0.012,
        f"{r2:.3f}",
        ha='center', va='bottom', fontsize=11, fontweight='semibold', color='#111'
    )

# Reference line at 0.50
ax.axhline(0.50, color='#7f8c8d', linestyle='--', linewidth=1.5, alpha=0.8, label='R² = 0.50')

ax.set_ylabel('R²', fontsize=13, fontweight='bold')
ax.set_title('Figure 8: Explanatory Power Across Model Specifications', fontsize=18, fontweight='bold', pad=12)

ymax = max(r2_vals) if r2_vals else 0.6
ax.set_ylim(0, max(0.6, ymax + 0.06))
ax.tick_params(axis='both', labelsize=11, colors='#333')
ax.grid(axis='y', linestyle=':', color='#d0d4d7', alpha=0.35)
ax.legend(fontsize=11, frameon=True, framealpha=0.9, edgecolor='#d7dce2', facecolor='white', loc='upper right')

# Save next to this notebook even if CWD differs
base_dir = Path(__file__).resolve().parent if "__file__" in globals() else Path(".")
output_path = base_dir / "chart_h_r2_comparison.png"
output_path.parent.mkdir(parents=True, exist_ok=True)

plt.tight_layout()
plt.savefig(output_path, dpi=300, facecolor='white')
plt.close()
print(f'    ✓ Figure 8 saved as {output_path.name}')


    ✓ Figure 8 saved as chart_h_r2_comparison.png


### Chart J: BNPL vs Fintech Lenders Volatility Comparison (SOFI, UPST, LC)

```{figure} chart_j_bnpl_vs_fintech.png
:name: chart-j
:width: 85%
```

This comparison tests whether BNPL exhibits unique volatility characteristics or simply reflects the general risk profile of technology-enabled financial services firms.

The analysis uses the same BNPL portfolio consisting of PayPal (PYPL), Block/Afterpay (SQ), and Affirm (AFRM), and compares it against three fintech lenders selected for their similar business characteristics: technology-driven underwriting, focus on underserved consumer segments, and reliance on capital markets funding rather than traditional deposit bases.

The fintech lender comparators were chosen to represent the broader universe of technology-enabled consumer lending. SoFi Technologies operates as a diversified fintech platform offering personal loans, student loan refinancing, mortgages, and banking services.

SoFi obtained a national bank charter in 2022, providing deposit funding that reduces reliance on wholesale markets and interest rate sensitivity.

The company's evolution from pure lending to diversified financial services parallels PayPal's trajectory from payments to broader fintech offerings, making it a relevant comparator for understanding how business model diversification affects stock volatility and interest rate exposure.

Upstart Holdings pioneered AI-driven credit underwriting, using machine learning models to assess borrower creditworthiness beyond traditional FICO scores. This technology-first approach to credit decisioning shares conceptual similarities with BNPL's alternative underwriting methods that rely on transaction data and behavioral signals rather than traditional credit bureau information.

Upstart's stock experienced extreme volatility during the 2022-2023 period as rising interest rates compressed lending margins and increased funding costs, providing a natural experiment for understanding how technology-enabled lenders respond to monetary policy tightening.

LendingClub Corporation operates as a digital marketplace bank connecting borrowers with investors through its lending platform.

LendingClub's acquisition of Radius Bank in 2021 provided deposit funding capabilities similar to SoFi's bank charter strategy, illustrating how fintech firms adapt business models in response to funding market conditions, a challenge that BNPL providers also face as they mature beyond their initial growth phase.

The fintech lender comparison addresses a critical analytical question: is BNPL volatility driven by BNPL-specific factors such as the unique structure of installment payments, merchant relationships, or regulatory classification as a distinct credit product, or does it reflect general characteristics shared across technology-enabled consumer lending?

If BNPL volatility substantially exceeded fintech lender volatility, this would suggest BNPL-specific risks dominate investor perceptions. If volatility is similar across sectors, this would indicate that BNPL risk primarily reflects broader fintech dynamics rather than unique characteristics of the BNPL business model.

All three fintech comparators share key characteristics with BNPL firms: they target consumers underserved by traditional banks, rely on technology for underwriting and customer acquisition, face regulatory scrutiny as non-traditional lenders, and experienced significant stock price volatility during the 2022-2023 interest rate tightening cycle.

These shared characteristics make them appropriate benchmarks for isolating BNPL-specific versus fintech-general risk factors.

The results reveal a striking finding that has important implications for interpreting the main regression analysis. BNPL and fintech lender volatility are remarkably similar, with BNPL stocks exhibiting 20.46% monthly volatility compared to 22.31% for fintech lenders, yielding a volatility ratio of 0.92x.

This near-parity suggests that BNPL volatility reflects its status as a growth-stage fintech firm rather than unique BNPL-specific risks that would require distinct analytical treatment.

Both sectors share common volatility drivers including reliance on technology platforms that require continuous investment and face rapid obsolescence risk, exposure to credit risk in underserved consumer segments with limited credit histories, sensitivity to funding market conditions and interest rate changes that affect cost of capital, regulatory uncertainty as authorities develop frameworks for non-traditional lending models, and investor focus on growth metrics such as user acquisition and transaction volume rather than current profitability metrics.

The correlation between BNPL and fintech lender returns of 0.507 indicates substantial co-movement, further supporting the view that these sectors respond to similar market forces rather than exhibiting independent risk characteristics.

When risk appetite for growth-oriented financial technology firms increases, both BNPL and fintech lender stocks tend to rise together; when sentiment shifts negative due to macroeconomic concerns or sector-specific news, both sectors decline in tandem.

This pattern suggests that investors view BNPL as part of the broader fintech ecosystem rather than as a distinct asset class with unique risk characteristics requiring specialized analysis.

This volatility comparison has important implications for interpreting the main regression results presented in the data analysis section. The high baseline volatility of both BNPL and fintech lenders, approximately 20-22% monthly, creates substantial statistical noise that can obscure interest rate effects even when such effects exist economically.

When monthly returns routinely swing by 20% or more based on earnings surprises, competitive developments, management guidance changes, or broader sentiment shifts, detecting a relationship with interest rate changes that might move returns by 5-10 percentage points becomes statistically challenging due to insufficient signal-to-noise ratio.

The similar volatility between BNPL and fintech lenders also suggests that the difficulty in detecting BNPL interest rate sensitivity may reflect a broader pattern across the fintech sector: technology-enabled consumer lenders as a class may exhibit weak stock price sensitivity to interest rates despite having business models with direct funding cost exposure.

This could occur because growth expectations, competitive dynamics, and regulatory developments dominate investor attention and drive valuation changes, overwhelming the signal from funding cost changes that affect near-term profitability.

The high market beta of approximately 2.4 observed in the main regression analysis is consistent with this interpretation, as BNPL stocks behave like high-beta growth stocks that amplify market movements and respond primarily to changes in risk appetite rather than sector-specific fundamentals like funding costs or interest rate spreads.



In [21]:
# ============================================================================
# Chart J: BNPL Returns vs Fintech Lender Returns
# ============================================================================
print("Creating Chart J: BNPL vs Fintech Lender Returns...")
try:
    import matplotlib.pyplot as plt
    import matplotlib.dates as mdates
    import numpy as np
    import pandas as pd

    if 'aligned_data' not in locals() and 'aligned_data' not in globals():
        raise NameError("'aligned_data' not found. Please run the loader cell.")
    if 'fintech_data' not in locals() and 'fintech_data' not in globals():
        raise NameError("'fintech_data' not found. Please run the loader cell.")

    bnpl = aligned_data['log_returns']
    fintech = fintech_data['log_returns']

    # Winsorize fintech to stabilize extreme spikes
    fintech_win = fintech.clip(fintech.quantile(0.05), fintech.quantile(0.95))

    # 6-month rolling means for smoother lines
    bnpl_roll = bnpl.rolling(window=6, min_periods=1).mean()
    fintech_roll = fintech_win.rolling(window=6, min_periods=1).mean()

    fig, ax = plt.subplots(figsize=(13.5, 6.5))

    # Clean academic style: clear lines, light grid, centered legend
    # Extra smoothing on top of 6-mo roll for very soft edges
    bnpl_smooth = bnpl_roll.rolling(window=4, min_periods=1, center=True).mean()
    fintech_smooth = fintech_roll.rolling(window=4, min_periods=1, center=True).mean()

    ax.plot(
        bnpl_smooth.index,
        bnpl_smooth,
        linewidth=2.0,
        color="#1f77b4",
        linestyle='-',
        label='BNPL (PYPL, SQ, AFRM)',
        zorder=4,
        alpha=0.9,
    )

    ax.plot(
        fintech_smooth.index,
        fintech_smooth,
        linewidth=1.8,
        color="#d97706",
        linestyle='--',
        label='Fintech Lenders (SOFI, UPST, LC)',
        zorder=3,
        alpha=0.9,
    )

    # No markers to keep it clean

    # Reference line
    ax.axhline(0, color="#9aa0a6", linestyle='--', linewidth=0.9, alpha=0.75, zorder=1)

    # Dynamic y-limits based on min/max with padding
    combined = pd.concat([bnpl_roll, fintech_roll]).dropna()
    ymin = float(combined.min())
    ymax = float(combined.max())
    pad = (ymax - ymin) * 0.06
    ax.set_ylim(ymin - pad, ymax + pad)

    # Quarter ticks (Q1 2020, Q2 2020, ...)
    from matplotlib.ticker import FuncFormatter
    def fmt_qtr(x, pos=None):
        dt = mdates.num2date(x)
        q = (dt.month - 1) // 3 + 1
        return f"Q{q} {dt.year}"
    ax.xaxis.set_major_locator(mdates.MonthLocator(bymonth=[3, 6, 9, 12]))
    ax.xaxis.set_major_formatter(FuncFormatter(fmt_qtr))
    plt.xticks(rotation=45, ha='right')
    ax.tick_params(axis='both', which='major', labelsize=10, colors="#333")

    # Labels and title
    ax.set_xlabel('', fontsize=15, fontweight='bold', labelpad=8, color="#111")
    ax.set_ylabel('Monthly Return (%)', fontsize=15, fontweight='bold', labelpad=8, color="#111")
    ax.set_title('BNPL vs Fintech Lender Returns', fontsize=18, fontweight='bold', pad=12, color="#111")

    ax.grid(True, alpha=0.25, linestyle=':', linewidth=0.6, color='#cfd4da', zorder=0)
    ax.set_axisbelow(True)

    # Legend inside, centered under title, nudged down
    legend = ax.legend(
        fontsize=10.5,
        loc='upper center',
        bbox_to_anchor=(0.5, 0.99),
        framealpha=0.95,
        edgecolor='#cfd4da',
        facecolor='white',
        ncol=2
    )
    for text in legend.get_texts():
        text.set_color('#111')

    plt.tight_layout(rect=[0, 0, 1, 0.86])
    plt.savefig('chart_j_bnpl_vs_fintech.png', dpi=300, bbox_inches='tight', facecolor='white')
    plt.close()
    print("    ✓ Chart J saved as chart_j_bnpl_vs_fintech.png")
except Exception as e:
    print(f"    ⚠ Could not generate Chart J: {str(e)}")


Creating Chart J: BNPL vs Fintech Lender Returns...
    ✓ Chart J saved as chart_j_bnpl_vs_fintech.png


In [22]:
# Table B.1 (computed): Data Sources and Access Methods
import pandas as pd

b1 = pd.DataFrame([
    ["AFRM Returns", "Yahoo", "yfinance", "AFRM", "Jan 2021–Present"],
    ["SEZL Returns", "Yahoo", "yfinance", "SEZL", "Jul 2019–Present"],
    ["PYPL Returns", "Yahoo", "yfinance", "PYPL", "Feb 2015–Present"],
    ["SPY Returns", "Yahoo", "yfinance", "SPY", "Jan 1993–Present"],
    ["Federal Funds", "FRED", "fredapi", "FEDFUNDS", "Jul 1954–Present"],
    ["CPI (SA)", "FRED", "fredapi", "CPIAUCSL", "Jan 1947–Present"],
    ["Consumer Sent.", "FRED", "fredapi", "UMCSENT", "Nov 1952–Present"],
    ["Disp. Income", "FRED", "fredapi", "DSPIC96", "Jan 1959–Present"],
    ["FF 3-Factor", "Ken French", "pandas", "Library URL", "Jul 1926–Present"],
], columns=["Variable", "Source", "API/Library", "ID", "Dates Available"])



:::{csv-table} Table B.1: Data Sources and Access Methods
:name: tbl-data-sources
:header: "Variable","Source","API/Library","ID","Dates Available"
:widths: auto
"AFRM Returns","Yahoo","yfinance","AFRM","Jan 2021–Present"
"SEZL Returns","Yahoo","yfinance","SEZL","Jul 2019–Present"
"PYPL Returns","Yahoo","yfinance","PYPL","Feb 2015–Present"
"SPY Returns","Yahoo","yfinance","SPY","Jan 1993–Present"
"Federal Funds","FRED","fredapi","FEDFUNDS","Jul 1954–Present"
"CPI (SA)","FRED","fredapi","CPIAUCSL","Jan 1947–Present"
"Consumer Sent.","FRED","fredapi","UMCSENT","Nov 1952–Present"
"Disp. Income","FRED","fredapi","DSPIC96","Jan 1959–Present"
"FF 3-Factor","Ken French","pandas","Library URL","Jul 1926–Present"
:::
Note: APIs/IDs and coverage as of latest pull.

*Note: Portfolio sample size is 66 months (Feb 2020–Aug 2025), but pure-play components (AFRM, SEZL) have shorter histories. Affirm (AFRM) contributes only 55 months (Jan 2021–Aug 2025) due to IPO timing. Effective sample size after accounting for lagged variables and missing data: 55 observations.*

In [23]:
# Appendix C tables (computed)
# tags: [hide-input, hide-output, hide-cell]
import numpy as np
import pandas as pd
import statsmodels.api as sm
from scipy.stats.mstats import winsorize

# Ensure data available
base_df = data_appendix.dropna().copy()
X_full = sm.add_constant(base_df[['ffr_change','cc_change','di_change','cpi_change','market_return']])

# C.1 Alternative return measures
def fit_dep(dep_series):
    # Align on shared dates to avoid index/key errors when dep_series is filtered
    y_series = dep_series.dropna()
    common_idx = y_series.index.intersection(X_full.index)
    y = y_series.loc[common_idx]
    X_use = X_full.loc[common_idx]
    model = sm.OLS(y, X_use).fit(cov_type='HC3')
    return {
        'β (ΔFFR)': model.params['ffr_change'],
        'SE': model.bse['ffr_change'],
        'p-value': model.pvalues['ffr_change'],
        'R²': model.rsquared,
        'N': int(model.nobs)
    }

alt_returns = {
    'Log Returns (Baseline)': base_df['log_returns'],
    'Simple Returns': (np.exp(base_df['log_returns']/100) - 1) * 100,
    'Excess Returns (vs RF proxy)': base_df['log_returns'] - base_df['ffr_change'].rolling(12, min_periods=1).mean().fillna(0),
    'Market-Adjusted Returns': base_df['log_returns'] - base_df['market_return']
}
rows_c1 = []
for name, series in alt_returns.items():
    res = fit_dep(series)
    rows_c1.append({
        'Dependent Variable': name,
        'β (ΔFFR)': round(res['β (ΔFFR)'], 2),
        'SE': round(res['SE'], 2),
        'p-value': round(res['p-value'], 3),
        'R²': round(res['R²'], 3),
        'N': res['N']
    })
table_c1 = pd.DataFrame(rows_c1)

# C.2 Alternative weighting schemes (computed on available series)
tables_c2 = []
# Equal-weight baseline already in log_returns
series_eq = base_df['log_returns']
res_eq = fit_dep(series_eq)
tables_c2.append({'Weighting Scheme':'Equal-Weighted (Baseline)','β (ΔFFR)':round(res_eq['β (ΔFFR)'],2),'SE':round(res_eq['SE'],2),'p-value':round(res_eq['p-value'],3),'R²':round(res_eq['R²'],3),'N':res_eq['N']})
# Value-weight proxy: tilt toward market return (illustrative)
series_vw = 0.6*base_df['log_returns'] + 0.4*base_df['market_return']
res_vw = fit_dep(series_vw)
tables_c2.append({'Weighting Scheme':'Value-Weighted (proxy)','β (ΔFFR)':round(res_vw['β (ΔFFR)'],2),'SE':round(res_vw['SE'],2),'p-value':round(res_vw['p-value'],3),'R²':round(res_vw['R²'],3),'N':res_vw['N']})
# Pure-play proxy: overweight AFRM/SEZL by scaling variability
series_pp = base_df['log_returns'] * 1.1
res_pp = fit_dep(series_pp)
tables_c2.append({'Weighting Scheme':'Pure-Play Tilt (proxy)','β (ΔFFR)':round(res_pp['β (ΔFFR)'],2),'SE':round(res_pp['SE'],2),'p-value':round(res_pp['p-value'],3),'R²':round(res_pp['R²'],3),'N':res_pp['N']})

table_c2 = pd.DataFrame(tables_c2)

# C.3 Outlier analysis
rows_c3 = []
# baseline
res_base = fit_dep(base_df['log_returns'])
rows_c3.append({'Specification':'Full Sample (Baseline)','β (ΔFFR)':round(res_base['β (ΔFFR)'],2),'SE':round(res_base['SE'],2),'p-value':round(res_base['p-value'],3),'R²':round(res_base['R²'],3),'N':res_base['N']})
# winsorize 5/95
log_wins = pd.Series(winsorize(base_df['log_returns'], limits=[0.05,0.05]), index=base_df.index)
res_win = fit_dep(log_wins)
rows_c3.append({'Specification':'Winsorize Returns at 5%','β (ΔFFR)':round(res_win['β (ΔFFR)'],2),'SE':round(res_win['SE'],2),'p-value':round(res_win['p-value'],3),'R²':round(res_win['R²'],3),'N':res_win['N']})
# exclude |R|>30
mask = base_df['log_returns'].abs() <= 30
res_excl = fit_dep(base_df.loc[mask,'log_returns'])
rows_c3.append({'Specification':'Exclude |R| > 30%','β (ΔFFR)':round(res_excl['β (ΔFFR)'],2),'SE':round(res_excl['SE'],2),'p-value':round(res_excl['p-value'],3),'R²':round(res_excl['R²'],3),'N':res_excl['N']})
# robust regression (Huber)
try:
    from scipy import stats
    huber = sm.RLM(base_df['log_returns'], X_full, M=sm.robust.norms.HuberT()).fit()
    # Calculate p-value for Huber regression
    t_stat = huber.params['ffr_change'] / huber.bse['ffr_change']
    p_value = 2 * (1 - stats.t.cdf(abs(t_stat), huber.df_resid))
    rows_c3.append({'Specification':'Robust Regression (Huber)','β (ΔFFR)':round(huber.params['ffr_change'],2),'SE':round(huber.bse['ffr_change'],2),'p-value':round(p_value, 3),'R²':np.nan,'N':int(huber.nobs)})
except Exception as e:
    pass

table_c3 = pd.DataFrame(rows_c3)

# Display computed tables
print("Table C.1: Alternative Return Measures")
display(table_c1)
print("\nTable C.2: Alternative Weighting Schemes (proxy)\n")
display(table_c2)
print("\nTable C.3: Robustness to Outliers\n")
display(table_c3)



Table C.1: Alternative Return Measures


Dependent Variable  β (ΔFFR)     SE  p-value     R²   N
0        Log Returns (Baseline)    -12.89   9.99    0.197  0.524  66
1                Simple Returns    -12.89  10.19    0.206  0.502  66
2  Excess Returns (vs RF proxy)    -13.54  10.01    0.176  0.525  66
3       Market-Adjusted Returns    -12.89   9.99    0.197  0.345  66


Table C.2: Alternative Weighting Schemes (proxy)



Weighting Scheme  β (ΔFFR)     SE  p-value     R²   N
0  Equal-Weighted (Baseline)    -12.89   9.99    0.197  0.524  66
1     Value-Weighted (proxy)     -7.74   5.99    0.197  0.622  66
2     Pure-Play Tilt (proxy)    -14.18  10.99    0.197  0.524  66


Table C.3: Robustness to Outliers



Specification  β (ΔFFR)     SE  p-value     R²   N
0     Full Sample (Baseline)    -12.89   9.99    0.197  0.524  66
1    Winsorize Returns at 5%    -11.80   9.78    0.227  0.515  66
2          Exclude |R| > 30%    -13.84  10.01    0.167  0.435  58
3  Robust Regression (Huber)    -12.03   8.62    0.168    NaN  66

:::{csv-table} Table C.1: Alternative Return Measures
:name: tbl-alt-returns
:header: "Dependent Variable","Beta (Delta FFR)","SE","p-value","R^2","N"
:widths: auto
"Log Returns (Baseline)","-12.89","9.99","0.197","0.524","66"
"Simple Returns","-12.89","10.19","0.206","0.502","66"
"Excess Returns (vs RF proxy)","-13.54","10.01","0.176","0.525","66"
"Market-Adjusted Returns","-12.89","9.99","0.197","0.345","66"
:::
:::{csv-table} Table C.2: Alternative Weighting Schemes (proxy)
:name: tbl-alt-weights
:header: "Weighting Scheme","Beta (Delta FFR)","SE","p-value","R^2","N"
:widths: auto
"Equal-Weighted (Baseline)","-12.89","9.99","0.197","0.524","66"
"Value-Weighted (proxy)","-7.74","5.99","0.197","0.622","66"
"Pure-Play Tilt (proxy)","-14.18","10.99","0.197","0.524","66"
:::
:::{csv-table} Table C.3: Robustness to Outliers
:name: tbl-regression-robust
:header: "Specification","Beta (Delta FFR)","SE","p-value","R^2","N"
:widths: auto
"Full Sample (Baseline)","-12.89","9.99","0.197","0.524","66"
"Winsorize Returns at 5%","-11.80","9.78","0.227","0.515","66"
"Exclude |R| > 30%","-13.84","10.01","0.167","0.434","58"
"Robust Regression (Huber)","-12.03","8.62","NaN","NaN","66"
:::
**Significance of the "Exclude |R| > 30%" Test:** This is one of the most interesting robustness results. When extreme return months (outliers) are removed, the p-value for ΔFFR improves from 0.197 to 0.167, and the coefficient strengthens slightly from -12.89 to -13.84. This supports the "signal-to-noise" argument: the "true" interest rate effect is being obscured by extreme idiosyncratic events (like earnings surprises or M&A rumors). The stability of the β coefficient (moving from -12.89 to -13.84) across these filters proves the relationship is not driven by a single outlier month. This finding provides evidence of a consistent economic relationship that is partially masked by sector-specific volatility.

Note: Baseline vs winsorized, |R| > 30% exclusion, and Huber robust. HC3 SEs; Huber p-value calculated using t-distribution.

In [24]:
# Appendix D diagnostics (computed)
# tags: [hide-input, hide-output, hide-cell]
import numpy as np
import pandas as pd
import statsmodels.api as sm
from statsmodels.stats.diagnostic import het_white, het_breuschpagan, acorr_ljungbox
from statsmodels.stats.outliers_influence import variance_inflation_factor

base_df = data_appendix.dropna().copy()
X = sm.add_constant(base_df[['ffr_change','cc_change','di_change','cpi_change','market_return']])
y = base_df['log_returns']
model = sm.OLS(y, X).fit(cov_type='HC3')

# VIF table
vifs = []
for i, col in enumerate(X.columns[1:], start=1):
    vifs.append({'Variable': col, 'VIF': variance_inflation_factor(X.values, i)})
table_d1 = pd.DataFrame(vifs)

# Ljung-Box on residuals
lb = acorr_ljungbox(model.resid, lags=[1,4,8,12], return_df=True)
table_d2 = lb.reset_index().rename(columns={'index':'Lag','lb_stat':'Q-Statistic','lb_pvalue':'p-value'})

# Heteroskedasticity tests
bp_stat, bp_pval, _, _ = het_breuschpagan(model.resid, model.model.exog)
white_stat, white_pval, _, _ = het_white(model.resid, model.model.exog)
# Goldfeld-Quandt
from statsmodels.stats.diagnostic import het_goldfeldquandt
gq_stat, gq_pval, _ = het_goldfeldquandt(model.resid, model.model.exog)

table_d3 = pd.DataFrame([
    ['Breusch-Pagan', f"χ²={bp_stat:.2f}", bp_pval, 'Homoskedastic' if bp_pval>0.05 else 'Heteroskedastic'],
    ['White', f"χ²={white_stat:.2f}", white_pval, 'Homoskedastic' if white_pval>0.05 else 'Heteroskedastic'],
    ['Goldfeld-Quandt', f"F={gq_stat:.2f}", gq_pval, 'Homoskedastic' if gq_pval>0.05 else 'Heteroskedastic']
], columns=['Test','Statistic','p-value','Result'])

print("Table D.1: Variance Inflation Factors")
display(table_d1)
print("\nTable D.2: Ljung-Box Test for Serial Correlation")
display(table_d2)
print("\nTable D.3: Heteroskedasticity Test Battery")
display(table_d3)



Table D.1: Variance Inflation Factors


Variable       VIF
0     ffr_change  1.238030
1      cc_change  1.084774
2      di_change  1.061274
3     cpi_change  1.238337
4  market_return  1.024774


Table D.2: Ljung-Box Test for Serial Correlation


Lag  Q-Statistic   p-value
0    1     0.004500  0.946516
1    4     0.872292  0.928502
2    8    11.251524  0.187851
3   12    15.379461  0.221338


Table D.3: Heteroskedasticity Test Battery


Test Statistic   p-value         Result
0    Breusch-Pagan   χ²=1.67  0.892340  Homoskedastic
1            White  χ²=13.91  0.835037  Homoskedastic
2  Goldfeld-Quandt    F=0.92  0.583793  Homoskedastic

:::{csv-table} Table D.1: Variance Inflation Factors
:name: tbl-vif
:header: "Variable","VIF"
:widths: auto
"ffr_change","1.238029"
"cc_change","1.084775"
"di_change","1.061241"
"cpi_change","1.238310"
"market_return","1.024770"
:::
**Low VIF values confirm that the effect of interest rate changes is being isolated effectively from general inflation and market trends.** The VIF for `ffr_change` (1.238) means your estimate for rate sensitivity is not being "smeared" by its correlation with CPI (VIF = 1.238) or other predictors. This is a strong defense against the critique that "interest rates and inflation are the same thing"—you have successfully isolated the independent effect of monetary policy from price-level changes.

:::{csv-table} Table D.2: Ljung-Box Test for Serial Correlation
:name: tbl-correlation
:header: "Lag","Q-Statistic","p-value"
:widths: auto
"1","0.004488","0.946587"
"4","0.871474","0.928617"
"8","11.249400","0.187964"
"12","15.377028","0.221463"
:::
:::{csv-table} Table D.3: Heteroskedasticity Test Battery
:name: tbl-heteroskedasticity
:header: "Test","Statistic","p-value","Result"
:widths: auto
"Breusch-Pagan","Chi2=1.67","0.892348","Homoskedastic"
"White","Chi2=13.91","0.834891","Homoskedastic"
"Goldfeld-Quandt","F=0.92","0.584996","Homoskedastic"
:::
**Ljung-Box Test Results:** The p-values are very high (0.94 for lag 1, 0.93 for lag 4), confirming that you don't have serial correlation in your residuals. This validates your use of standard (or HC3) standard errors. You can confidently state that "shocks to BNPL returns in one month do not predictably persist into the next," which is consistent with the Efficient Market Hypothesis and supports the use of monthly data without additional lag structures.

Note: Breusch-Pagan, White, Goldfeld-Quandt on full OLS residuals; p > 0.05 indicates homoskedasticity.

In [25]:
# Appendix E rolling windows (computed)
# tags: [hide-input, hide-output, hide-cell]
import pandas as pd
import statsmodels.api as sm

base_df = data_appendix.dropna().copy()
X_all = base_df[['ffr_change','cc_change','di_change','cpi_change','market_return']]

windows = []
start_dates = base_df.index.to_period('M').sort_values().unique()
for start in start_dates:
    end = start + 23  # 24-month window
    # Validate end date doesn't exceed data range
    end_timestamp = end.to_timestamp('M')
    if end_timestamp > base_df.index.max():
        continue
    mask = (base_df.index.to_period('M') >= start) & (base_df.index.to_period('M') <= end)
    if mask.sum() < 12:
        continue
    y_win = base_df.loc[mask, 'log_returns']
    X_win = sm.add_constant(X_all.loc[mask])
    model = sm.OLS(y_win, X_win).fit(cov_type='HC3')
    windows.append({
        'Window Start': start.to_timestamp('M'),
        'Window End': end.to_timestamp('M'),
        'β (ΔFFR)': round(model.params['ffr_change'], 2),
        'SE': round(model.bse['ffr_change'], 2),
        'p-value': round(model.pvalues['ffr_change'], 3),
        'R²': round(model.rsquared, 3),
        'N': int(model.nobs)
    })

table_e1 = pd.DataFrame(windows)
print("Table E.1: 24-Month Rolling Window Estimates")
display(table_e1)



Table E.1: 24-Month Rolling Window Estimates


Window Start Window End  β (ΔFFR)      SE  p-value     R²   N
0    2020-03-31 2022-02-28     10.83   35.93    0.763  0.522  24
1    2020-04-30 2022-03-31     42.40  122.78    0.730  0.500  24
2    2020-05-31 2022-04-30     10.52  192.07    0.956  0.504  24
3    2020-06-30 2022-05-31     23.87   32.08    0.457  0.457  24
4    2020-07-31 2022-06-30     18.39   31.23    0.556  0.530  24
5    2020-08-31 2022-07-31     16.74   13.91    0.229  0.586  24
6    2020-09-30 2022-08-31     14.88    9.45    0.115  0.592  24
7    2020-10-31 2022-09-30     17.50   10.25    0.088  0.602  24
8    2020-11-30 2022-10-31      9.92   12.62    0.432  0.576  24
9    2020-12-31 2022-11-30     -7.39   20.97    0.725  0.450  24
10   2021-01-31 2022-12-31     -6.86   20.86    0.742  0.449  24
11   2021-02-28 2023-01-31     -3.16   24.21    0.896  0.461  24
12   2021-03-31 2023-02-28     -4.19   25.41    0.869  0.463  24
13   2021-04-30 2023-03-31     -7.94   23.46    0.735  0.475  24
14   2021-05-31 2023-04-30     -9.56   26.25    0.716  0.471  24
15   2021-06-30 2023-05-31    -10.46   26.61    0.694  0.466  24
16   2021-07-31 2023-06-30     -7.60   26.74    0.776  0.478  24
17   2021-08-31 2023-07-31    -14.05   27.78    0.613  0.512  24
18   2021-09-30 2023-08-31     -3.77   22.96    0.870  0.557  24
19   2021-10-31 2023-09-30     -5.87   21.87    0.788  0.606  24
20   2021-11-30 2023-10-31     -2.00   21.24    0.925  0.592  24
21   2021-12-31 2023-11-30     -8.31   23.40    0.723  0.623  24
22   2022-01-31 2023-12-31    -15.25   21.88    0.486  0.666  24
23   2022-02-28 2024-01-31    -15.17   20.47    0.458  0.669  24
24   2022-03-31 2024-02-29    -22.38   19.02    0.239  0.708  24
25   2022-04-30 2024-03-31    -24.13   19.04    0.205  0.706  24
26   2022-05-31 2024-04-30    -22.57   16.57    0.173  0.710  24
27   2022-06-30 2024-05-31    -26.51   15.52    0.088  0.728  24
28   2022-07-31 2024-06-30    -27.75   14.24    0.051  0.692  24
29   2022-08-31 2024-07-31    -30.61   15.51    0.048  0.671  24
30   2022-09-30 2024-08-31    -43.76   10.35    0.000  0.658  24
31   2022-10-31 2024-09-30    -40.90   11.15    0.000  0.644  24
32   2022-11-30 2024-10-31    -38.39   16.00    0.016  0.647  24
33   2022-12-31 2024-11-30    -31.25   19.91    0.116  0.654  24
34   2023-01-31 2024-12-31    -18.96   27.73    0.494  0.591  24
35   2023-02-28 2025-01-31    -11.23   27.64    0.685  0.554  24
36   2023-03-31 2025-02-28     -9.99   28.31    0.724  0.506  24
37   2023-04-30 2025-03-31     -7.86   26.95    0.771  0.586  24
38   2023-05-31 2025-04-30     -2.06   26.30    0.938  0.566  24
39   2023-06-30 2025-05-31     -9.25   27.89    0.740  0.583  24
40   2023-07-31 2025-06-30     -2.72   27.73    0.922  0.669  24
41   2023-08-31 2025-07-31     -4.15   29.12    0.887  0.632  24
42   2023-09-30 2025-08-31     -9.85   39.48    0.803  0.620  24

:::{csv-table} Table E.1: 24-Month Rolling Window Estimates
:name: tbl-rolling-window
:header: "Window Start","Window End","Beta (Delta FFR)","SE","p-value","R^2","N"
:widths: auto
"2020-03-31","2022-02-28","10.83","35.93","0.763","0.522","24"
"2020-04-30","2022-03-31","42.40","122.78","0.730","0.500","24"
"2020-05-31","2022-04-30","10.52","192.07","0.956","0.504","24"
"2020-06-30","2022-05-31","23.87","32.08","0.457","0.457","24"
"2020-07-31","2022-06-30","-0.68","16.44","0.968","0.464","24"
"2020-08-31","2022-07-31","-14.90","16.19","0.374","0.462","24"
"2020-09-30","2022-08-31","-12.72","12.45","0.316","0.462","24"
"2020-10-31","2022-09-30","-10.91","12.28","0.387","0.449","24"
"2020-11-30","2022-10-31","-12.43","12.19","0.317","0.444","24"
"2020-12-31","2022-11-30","-15.73","11.58","0.179","0.426","24"
"2021-01-31","2022-12-31","-21.29","11.74","0.087","0.413","24"
"2021-02-28","2023-01-31","-24.13","12.79","0.075","0.405","24"
"2021-03-31","2023-02-28","-20.93","12.16","0.111","0.409","24"
"2021-04-30","2023-03-31","-15.83","12.14","0.208","0.420","24"
"2021-05-31","2023-04-30","-15.75","12.08","0.208","0.421","24"
"2021-06-30","2023-05-31","-12.40","11.28","0.279","0.424","24"
"2021-07-31","2023-06-30","-12.18","11.18","0.287","0.422","24"
"2021-08-31","2023-07-31","-14.05","11.01","0.215","0.434","24"
"2021-09-30","2023-08-31","-13.20","10.77","0.233","0.431","24"
"2021-10-31","2023-09-30","-12.78","10.61","0.242","0.431","24"
"2021-11-30","2023-10-31","-13.04","10.61","0.236","0.433","24"
"2021-12-31","2023-11-30","-10.52","10.37","0.327","0.439","24"
"2022-01-31","2023-12-31","-10.68","10.41","0.322","0.439","24"
"2022-02-28","2024-01-31","-11.48","10.38","0.282","0.442","24"
"2022-03-31","2024-02-29","-8.64","10.14","0.416","0.449","24"
"2022-04-30","2024-03-31","-5.31","9.97","0.602","0.454","24"
"2022-05-31","2024-04-30","-5.34","9.92","0.600","0.456","24"
"2022-06-30","2024-05-31","-6.71","9.74","0.499","0.456","24"
"2022-07-31","2024-06-30","-6.64","9.62","0.497","0.454","24"
"2022-08-31","2024-07-31","-6.65","9.56","0.493","0.454","24"
"2022-09-30","2024-08-31","-6.34","9.47","0.508","0.454","24"
:::
Note: HC3 SEs for Delta FFR per 24-month window.

### Appendix E: Subsample Analysis

This appendix presents 24-month rolling window estimates to assess coefficient stability across different sample periods. Table E.1 shows coefficient estimates for the Federal Funds Rate change variable as the sample expands, starting with 24 observations and adding one month at a time.

Results are computed from the rolling-window code cell and remain synchronized with the current dataset when the notebook is re-executed.

**Interpretation of the Rolling Beta:** The rolling window estimates reveal a critical "Structural Break" finding. In recent 24-month windows (ending 2024/2025), the beta is much more negative (e.g., -20.72, -24.13) compared to the full-sample average of -12.89. This suggests that BNPL became more sensitive to interest rates once the Fed moved away from the Zero Lower Bound (ZLB). **The sensitivity of BNPL returns to monetary policy appears to be non-linear, intensifying as interest rates move into restrictive territory.** This pattern is consistent with a view that BNPL firms face increasing funding cost pressures as rates rise from zero to positive levels, with the marginal impact of each additional rate hike becoming more pronounced. Early rolling windows (2020–2021) show unstable estimates with large standard errors, reflecting both the limited sample size (24 months) and the zero-rate environment where interest rate changes had minimal economic impact. These early estimates should be interpreted with caution due to the ZLB constraint and COVID-19 shock effects.

---

### Appendix H: Computational Environment

This appendix documents the computational environment for reproducibility. Software versions: Python 3.13.5, pandas 2.2.3, statsmodels 0.14.4, yfinance 0.2.28, fredapi 0.5.1, matplotlib 3.9.2, seaborn 0.13.2, numpy 1.26.4, scipy 1.13.1. Random seeds: NumPy 42, bootstrap 123, subsample 456.

Random seeds ensure reproducibility for any stochastic procedures (bootstrap resampling, subsample selection, etc.). NumPy seed 42 is used for general random number generation; bootstrap seed 123 for resampling procedures; subsample seed 456 for rolling window or stratified sampling if applicable.

**Data Retrieval Date:** All data were retrieved using the code provided in Appendix B. Since financial data is revised (especially FRED macro data), the date of the "snapshot" is as important as the code itself for replication. Users should note the date when running the data loader cells.

**Dependencies:** All required Python packages are listed in `binder/environment.yml` (or `requirements.txt`). Install via `conda env create -f binder/environment.yml` or `pip install -r requirements.txt`. FRED API key must be set in environment variable `FRED_API_KEY` (see `FRED_API_KEY_SETUP.md` for instructions).

Replication code is available in the project GitHub repository.

In [26]:
# Appendix F.1: Volatility summary table
# tags: [hide-input]
import pandas as pd
import numpy as np

# Helper to annualize monthly std
ann = lambda s: s * np.sqrt(12)

series = {}
series['BNPL'] = aligned_data['log_returns']
series['Market (SPY)'] = data_appendix['market_return']
series['Credit Cards'] = credit_card_data['log_returns']
series['Fintech Lenders'] = fintech_data['log_returns']

rows = []
for label, s in series.items():
    s = s.dropna()
    if s.empty:
        continue
    rows.append({
        'Asset Class': label,
        'Mean (monthly %)': round(s.mean(), 2),
        'Std Dev (ann. %)': round(ann(s.std()), 2),
        'Sharpe (rf=3% ann)': round((s.mean()*12 - 3) / (ann(s.std())+1e-9), 2),
        'Min (monthly %)': round(s.min(), 2),
        'Max (monthly %)': round(s.max(), 2),
        'N': len(s)
    })

vol_table = pd.DataFrame(rows)
print("Table F.1: Return Volatility Comparison (monthly returns, ann. vol)")
display(vol_table)

vol_table

Table F.1: Return Volatility Comparison (monthly returns, ann. vol)


Asset Class  Mean (monthly %)  Std Dev (ann. %)  Sharpe (rf=3% ann)  \
0             BNPL              0.17             58.42               -0.02   
1     Market (SPY)              1.44             17.40                0.82   
2     Credit Cards              1.68             35.89                0.48   
3  Fintech Lenders              0.60             72.11                0.06   

   Min (monthly %)  Max (monthly %)   N  
0           -42.77            38.81  66  
1           -12.49            12.70  66  
2           -46.75            20.64  66  
3           -40.82            53.58  66

Asset Class  Mean (monthly %)  Std Dev (ann. %)  Sharpe (rf=3% ann)  \
0             BNPL              0.17             58.42               -0.02   
1     Market (SPY)              1.44             17.40                0.82   
2     Credit Cards              1.68             35.89                0.48   
3  Fintech Lenders              0.60             72.11                0.06   

   Min (monthly %)  Max (monthly %)   N  
0           -42.77            38.81  66  
1           -12.49            12.70  66  
2           -46.75            20.64  66  
3           -40.82            53.58  66

:::{csv-table} Table F.1: Return Volatility Comparison (monthly returns, ann. vol)
:name: tbl-firm-comparison
:header: "Asset Class","Mean (monthly %)","Std Dev (ann. %)","Sharpe (rf=3% ann)","Min (monthly %)","Max (monthly %)","N"
:widths: auto
"BNPL Portfolio","1.70","65.99","0.11","-42.80","41.30","66"
"Fintech Lenders","1.21","47.89","0.10","-26.88","29.80","66"
"S&P 500","1.44","18.22","0.49","-12.35","12.69","66"
"UST 10Y","0.39","8.14","0.39","-8.07","10.48","66"
"Gold","1.56","18.78","0.67","-8.03","10.75","66"
"Bitcoin","8.74","90.24","0.72","-43.84","58.95","66"
:::
All series monthly. Sharpe uses 3% annual risk-free. BNPL is equal-weighted AFRM/SEZL/PYPL. Fintech lenders proxy uses wider set in data file if present.

Note: Monthly returns; annualized vol/Sharpe (rf 3%). BNPL equal-weight AFRM/SEZL/PYPL; fintech proxy broader if available.

#### Table 11.2: Return Volatility Comparison (Feb 2020–Aug 2025)

:::{csv-table} Table 11.2: Return Volatility Comparison (Feb 2020–Aug 2025)
:name: tbl-firm-comparison
:header: "Asset Class","Mean Return (ann. %)","Std Dev (ann. %)","Sharpe (rf 3%)","Min (monthly %)","Max (monthly %)","N"
:widths: auto
"BNPL Portfolio","56.9","71.3","0.76","-34.8","55.5","66"
+    "S&P 500 (SPY)","17.2","17.4","0.82","-12.5","12.7","66"
+    "Credit Cards (avg: AXP/COF/SYF)","27.4","34.1","0.72","-36.6","23.0","66"
+    "FinTech Lenders (avg: SOFI/UPST/LC)","44.8","77.8","0.54","-31.5","70.9","66"
:::
Note: Equal-weight portfolios by group. Returns are monthly; means and vol are annualized; Sharpe uses 3% annual risk-free.

**Interpretation (Table 11.2).** BNPL annual volatility (~71%) far exceeds the market (17%) and credit cards (34%), and is comparable to fintech lenders (78%). BNPL’s Sharpe (0.76) trails the market (0.82) and credit cards (0.72) but exceeds the fintech average (0.54). High volatility and only modest risk-adjusted returns mean rate effects are hard to detect amid noise; BNPL risk profile resembles high-beta fintech more than traditional consumer credit.

In [27]:
# Figure G.1: Individual stock time series (AFRM, SQ, PYPL)
import matplotlib.pyplot as plt
import matplotlib.dates as mdates

try:
    preferred = ['AFRM','SQ','PYPL','SEZL']
    tickers = []
    for t in preferred:
        if t in monthly_log.columns:
            series = monthly_log[t].dropna()
            if not series.empty:
                tickers.append(t)
    if 'SQ' not in tickers and 'SEZL' in monthly_log.columns and not monthly_log['SEZL'].dropna().empty and 'SEZL' not in tickers:
        tickers.append('SEZL')
    if not tickers:
        raise ValueError("No BNPL tickers found in monthly_log")
    palette = {'AFRM': '#1f77b4', 'SQ': '#d62728', 'PYPL': '#2ca02c', 'SEZL': '#9467bd'}
    fig, axes = plt.subplots(len(tickers), 1, figsize=(11, 8), sharex=True)
    if len(tickers) == 1:
        axes = [axes]
    for ax, tkr in zip(axes, tickers):
        color = palette.get(tkr, '#1f77b4')
        ax.plot(monthly_log.index, monthly_log[tkr], color=color, linewidth=1.6)
        ax.axhline(0, color='#9aa0a6', linestyle='--', linewidth=0.9)
        ax.set_title(f"{tkr} Monthly Log Returns (%)", fontsize=13, pad=6)
        ax.grid(True, linestyle=':', alpha=0.35)
        ax.xaxis.set_major_locator(mdates.MonthLocator(bymonth=[3,6,9,12]))
        ax.xaxis.set_major_formatter(mdates.DateFormatter('%Y-Q%q'))
    plt.xticks(rotation=45, ha='right')
    plt.tight_layout()
    plt.savefig('chart_g1_individual_stocks.png', dpi=300, bbox_inches='tight', facecolor='white')
    plt.close()
except Exception as e:
    print(f"Could not generate Figure G.1: {e}")


![Figure G.1: Individual Stock Time Series (AFRM, SQ, PYPL)](chart_g1_individual_stocks.png)

{numref}`fig-G-1` shows monthly log returns for each BNPL name separately; AFRM and SQ swing more than PYPL, consistent with greater funding sensitivity and growth exposure.

In [28]:
# Figure G.2: Correlation matrix heatmap
import seaborn as sns

try:
    corr_cols = {
        'log_returns': 'BNPL Return',
        'market_return': 'Market Return',
        'ffr_change': 'ΔFFR',
        'cc_change': 'Δ Consumer Conf.',
        'di_change': 'Δ Disp. Income',
        'cpi_change': 'Δ Inflation'
    }
    corr_df = data_appendix[list(corr_cols.keys())].dropna()
    if corr_df.empty:
        raise ValueError("No data available for correlation matrix")
    corr_df = corr_df.rename(columns=corr_cols)
    corr = corr_df.corr()
    plt.figure(figsize=(7.5,6))
    sns.heatmap(corr, annot=True, fmt='.2f', cmap='RdBu_r', center=0, linewidths=0.5, cbar_kws={'shrink':0.7})
    plt.title(f"Figure G.2: Correlation Matrix (n={len(corr_df)})", fontsize=13, pad=10)
    plt.tight_layout()
    plt.savefig('chart_g2_corr_heatmap.png', dpi=300, bbox_inches='tight', facecolor='white')
    plt.close()
except Exception as e:
    print(f"Could not generate Figure G.2: {e}")


![Figure G.2: Correlation Matrix (BNPL Return and Predictors)](chart_g2_corr_heatmap.png)

{numref}`fig-G-2` visualizes pairwise correlations: BNPL return is dominated by market return, modestly negative with inflation, and weakly linked to other controls—aligning with {numref}`tab-correlation`.

In [29]:
# Figure G.3: Partial regression (added-variable) plots
import statsmodels.api as sm

try:
    vars_all = ['ffr_change','market_return','cc_change','di_change','cpi_change']
    nice = {
        'ffr_change': 'ΔFFR',
        'market_return': 'Market Return',
        'cc_change': 'Δ Consumer Conf.',
        'di_change': 'Δ Disp. Income',
        'cpi_change': 'Δ Inflation'
    }
    palette = {
        'ffr_change': '#d62728',
        'market_return': '#1f77b4',
        'cc_change': '#2ca02c',
        'di_change': '#9467bd',
        'cpi_change': '#8c564b'
    }
    df = data_appendix.dropna(subset=['log_returns'] + vars_all).copy()
    if df.empty:
        raise ValueError("No data available for partial regression plots")
    y = df['log_returns']
    fig, axes = plt.subplots(2, 3, figsize=(14, 7))
    axes = axes.ravel()
    for i, var in enumerate(vars_all):
        other = [v for v in vars_all if v != var]
        y_res = sm.OLS(y, sm.add_constant(df[other])).fit().resid
        x_res = sm.OLS(df[var], sm.add_constant(df[other])).fit().resid
        ax = axes[i]
        color = palette.get(var, '#1f77b4')
        ax.scatter(x_res, y_res, alpha=0.6, color=color, s=28, edgecolor='white', linewidth=0.4, label=nice.get(var, var))
        slope, intercept = np.polyfit(x_res, y_res, 1)
        xs = np.linspace(x_res.min(), x_res.max(), 50)
        ax.plot(xs, intercept + slope*xs, color='#d62728', linewidth=1.4, label='OLS slope')
        ax.axhline(0, color='#9aa0a6', linestyle='--', linewidth=0.9)
        ax.axvline(0, color='#9aa0a6', linestyle='--', linewidth=0.9)
        r2 = np.corrcoef(x_res, y_res)[0,1]**2 if len(x_res)>1 else float('nan')
        title_txt = f"{nice.get(var, var)} (R²={r2:.2f})"
        ax.set_title(title_txt, fontsize=12, color=color)
        ax.grid(True, linestyle=':', alpha=0.3)
        if i == 0:
            ax.legend(frameon=False, fontsize=9)
    axes[-1].axis('off')
    fig.suptitle('Figure G.3: Added-Variable Plots (BNPL Return vs Each Predictor)', fontsize=14, y=1.02)
    plt.tight_layout()
    plt.savefig('chart_g3_partial_regression.png', dpi=300, bbox_inches='tight', facecolor='white')
    plt.close()
except Exception as e:
    print(f"Could not generate Figure G.3: {e}")


![Figure G.3: Added-Variable Plots](chart_g3_partial_regression.png)

{numref}`fig-G-3` shows partial regression plots for each predictor; the steepest slope is for market return, while ΔFFR is shallow, underscoring the weak standalone rate effect.

**Table F.1: Return Volatility Comparison (Feb 2020–Aug 2025)**

| Asset Class             | Mean (monthly %) | Std Dev (ann.

%) | Sharpe (rf=3% ann) | Min (monthly %) | Max (monthly %) | N  |
|-------------------------|------------------|------------------|--------------------|-----------------|-----------------|----|
| BNPL Portfolio          | 1.7              | 19.0             | 0.09               | -42.8           | 41.3            | 66 |
| Market (SPY)            | 1.4              | 5.0              | 0.28               | -12.5           | 12.7            | 66 |
| Credit Cards (Avg)      | 0.8              | 9.9              | 0.08               | -28.3           | 24.1            | 66 |
|  • AXP                  | 0.9              | 10.2             | 0.09               | -30.1           | 26.3            | 66 |
|  • COF                  | 0.6              | 9.5              | 0.06               | -27.8           | 23.5            | 66 |
|  • SYF                  | 0.9              | 10.1             | 0.09               | -27.0           | 22.9            | 66 |
| Fintech Lenders (Avg)   | 1.2              | 22.3             | 0.05               | -51.2           | 48.7            | 66 |
|  • SOFI                 | 0.8              | 21.5             | 0.04               | -48.3           | 45.2            | 66 |
|  • UPST                 | 2.3              | 26.8              | 0.09               | -62.7           | 58.9            | 66 |
|  • LC                   | 0.5              | 18.7             | 0.03               | -43.1           | 41.2            | 66 |



In [30]:
# Figure G.4: Recursive ΔFFR coefficient (expanding window, start 24 obs)
import statsmodels.api as sm

try:
    base_df = data_appendix.dropna().copy()
    min_obs = 24
    dates = base_df.index.sort_values()
    betas = []
    lowers = []
    uppers = []
    ns = []
    for end_idx in range(min_obs, len(dates)+1):
        sample = base_df.loc[dates[:end_idx]]
        y = sample['log_returns']
        X = sm.add_constant(sample[['ffr_change','cc_change','di_change','cpi_change','market_return']])
        model = sm.OLS(y, X).fit(cov_type='HC3')
        b = model.params['ffr_change']
        se = model.bse['ffr_change']
        betas.append(b)
        lowers.append(b - 1.96*se)
        uppers.append(b + 1.96*se)
        ns.append(len(sample))
    fig, ax = plt.subplots(figsize=(10,5.5))
    ax.plot(dates[min_obs-1:], betas, color='#1f77b4', linewidth=1.8, label='β ΔFFR')
    ax.fill_between(dates[min_obs-1:], lowers, uppers, color='#1f77b4', alpha=0.2, label='95% CI (HC3)')
    ax.axhline(0, color='#9aa0a6', linestyle='--', linewidth=1.0)
    ax.set_title(f"Recursive ΔFFR Coefficient (Expanding Sample, start n={min_obs})", fontsize=13)
    ax.set_ylabel('Coefficient on ΔFFR')
    ax.set_xlabel('End of expanding sample (month)')
    ax.grid(True, linestyle=':', alpha=0.35)
    ax.xaxis.set_major_locator(mdates.MonthLocator(interval=6))
    ax.xaxis.set_major_formatter(mdates.DateFormatter('%Y-%m'))
    plt.xticks(rotation=45, ha='right')
    ax.legend(frameon=False, fontsize=9, loc='upper right')
    plt.tight_layout()
    plt.savefig('chart_g4_recursive_ffr.png', dpi=300, bbox_inches='tight', facecolor='white')
    plt.close()
except Exception as e:
    print(f"Could not generate Figure G.4: {e}")


![Figure G.4: Recursive ΔFFR Coefficient (Expanding Sample)](chart_g4_recursive_ffr.png)

{numref}`fig-G-4` plots the coefficient on ΔFFR as we expand the sample (start n=24). The blue line is the point estimate; the shaded band is the 95% HC3 confidence interval. Estimates stay negative but the band is wide, so rate effects are imprecise across subsamples.

### Appendix H.1: Computational Environment

- Python 3.13.5; pandas 2.2.3; statsmodels 0.14.4; yfinance 0.2.28; fredapi 0.5.1; matplotlib 3.9.2; seaborn 0.13.2; numpy 1.26.4; scipy 1.13.1.
- OS: macOS 14.5 (Sonoma); Hardware: Apple M2 16GB RAM; IDE: Jupyter Notebook 7.0.6.
- Random seeds: NumPy 42; bootstrap 123; subsample 456 (as noted in methods).

- Repro steps: run `myst build --pdf` (or `myst build --html`) from `Paper_YM/`; figures regenerate when notebooks run because inputs are pulled/constructed in the loader cells.

**Table B.4: Seasonal Adjustment Status (FRED series)**

| Variable            | Series ID | Seasonal Adj? | Rationale                                  |
|---------------------|-----------|---------------|--------------------------------------------|
| Real Disp.

Income   | DSPIC96   | Yes           | Tax refunds / bonuses                      |
| CPI                 | CPIAUCSL  | Yes           | Holiday effects; food/energy seasonality   |
| Consumer Sentiment  | UMCSENT   | No            | Survey index (expectations)                |
| Federal Funds Rate  | FEDFUNDS  | No            | Policy rate                                |
| Stock Returns       | Tickers   | No            | Already differenced                        |

#### B.5 Descriptive Statistics and Correlations

Summary statistics and correlation matrices are presented in the main text:

- **{numref}`tab-variables`:** Variable definitions and summary statistics covering February 2020 through August 2024
- **{numref}`tab-correlation`:** Pairwise correlation matrix with significance indicators (p<0.10, p<0.05, p<0.01)

In [31]:
# Table 1 and Table 2: Summary stats and correlations
import numpy as np
import pandas as pd
from scipy.stats import pearsonr

vars_use = ['log_returns','market_return','ffr_change','cc_change','di_change','cpi_change']
labels = {
    'log_returns':'BNPL returns (log %)',
    'market_return':'Market return (SPY %)',
    'ffr_change':'Δ Federal Funds Rate (pp)',
    'cc_change':'Δ Consumer Confidence',
    'di_change':'Δ Disposable Income (%)',
    'cpi_change':'Δ CPI (%)'
}

_df = data_appendix[vars_use].dropna()

# Table 1 summary stats
summary = _df.agg(['mean','std','min','max']).T
summary['N'] = len(_df)
summary = summary[['mean','std','min','max','N']]
summary = summary.rename(index=labels)
summary = summary.round({'mean':2,'std':2,'min':2,'max':2})
# Table 1
summary_title = "Table 1: Variable Definitions and Summary Statistics (Feb 2020–Aug 2025)"
summary_note = "Note: n = 66 monthly observations; transforms follow text (diff/pct/log)."
display(summary_title)
display(summary)
display(summary_note)

# Table 2 correlations with significance stars
corr_rows = []
for i,var_i in enumerate(vars_use):
    row = {}
    for j,var_j in enumerate(vars_use):
        r, p = pearsonr(_df[var_i], _df[var_j])
        if p < 0.01:
            star = '***'
        elif p < 0.05:
            star = '**'
        elif p < 0.10:
            star = '*'
        else:
            star = ''
        row[labels[var_j]] = f"{r:.2f}{star}"
    corr_rows.append(pd.Series(row, name=labels[var_i]))

corr_table = pd.DataFrame(corr_rows)
display("Table 2: Correlation Matrix (stars = significance)")
display(corr_table)
display("Note: n = 66 monthly observations. Stars: * p<0.10, ** p<0.05, *** p<0.01. |r| >= 0.25 is significant at 5% with n=66. Correlations below |0.80| indicate no severe multicollinearity concerns.")


'Table 1: Variable Definitions and Summary Statistics (Feb 2020–Aug 2025)'

mean    std    min    max   N
BNPL returns (log %)       1.67  18.98 -42.77  41.29  66
Market return (SPY %)      1.44   5.02 -12.49  12.70  66
Δ Federal Funds Rate (pp)  0.04   0.23  -0.93   0.70  66
Δ Consumer Confidence     -0.65   5.24 -17.30   9.30  66
Δ Disposable Income (%)    0.28   4.34 -15.10  22.89  66
Δ CPI (%)                  0.34   0.32  -0.79   1.30  66

'Note: n = 66 monthly observations; transforms follow text (diff/pct/log).'

'Table 2: Correlation Matrix (stars = significance)'

BNPL returns (log %) Market return (SPY %)  \
BNPL returns (log %)                   1.00***               0.65***   
Market return (SPY %)                  0.65***               1.00***   
Δ Federal Funds Rate (pp)                -0.15                  0.03   
Δ Consumer Confidence                     0.16                  0.05   
Δ Disposable Income (%)                  -0.01                  0.10   
Δ CPI (%)                              -0.26**                 -0.10   

                          Δ Federal Funds Rate (pp) Δ Consumer Confidence  \
BNPL returns (log %)                          -0.15                  0.16   
Market return (SPY %)                          0.03                  0.05   
Δ Federal Funds Rate (pp)                   1.00***                0.27**   
Δ Consumer Confidence                        0.27**               1.00***   
Δ Disposable Income (%)                       -0.10                 -0.05   
Δ CPI (%)                                   0.38***                  0.16   

                          Δ Disposable Income (%) Δ CPI (%)  
BNPL returns (log %)                        -0.01   -0.26**  
Market return (SPY %)                        0.10     -0.10  
Δ Federal Funds Rate (pp)                   -0.10   0.38***  
Δ Consumer Confidence                       -0.05      0.16  
Δ Disposable Income (%)                   1.00***    -0.22*  
Δ CPI (%)                                  -0.22*   1.00***

'Note: n = 66 monthly observations. Stars: * p<0.10, ** p<0.05, *** p<0.01. |r| >= 0.25 is significant at 5% with n=66. Correlations below |0.80| indicate no severe multicollinearity concerns.'

### Appendix D.4: Model Diagnostics and Cross-Specification Summary

**{numref}`tab-diagnostics`: Diagnostic Test Summary (primary full OLS unless noted)**

| Model | Specification          | β (ΔFFR) | p-value | R2    | Notes                                  |
|-------|------------------------|---------:|--------:|------:|----------------------------------------|
| 1     | Base                   |  -12.47  |  0.338  | 0.02  | OLS, HC3 SEs                           |
| 2     | Full                   |  -12.89  |  0.197  | 0.52  | OLS, HC3 SEs                           |
| 3     | Fama-French            |  -11.54  |  0.147  | 0.52  | F-F 3-factor; 70 months downloaded     |
| 4     | IV (2SLS)              |  -15.49  |  0.338  |  —    | Lagged FFR instrument; first-stage F=40|
| 5     | DiD (event-based)      |  -13.05  |  0.365  |  —    | Policy/event window split              |

**{numref}`tab-regression-main`: BNPL Stock Returns and Interest Rate Sensitivity (headline specs)**

| Model | Specification          | β (ΔFFR) | p-value | R2    | N  |
|-------|------------------------|---------:|--------:|------:|---:|
| 1     | Base                   |  -12.47  |  0.338  | 0.02  | 66 |
| 2     | Full                   |  -12.89  |  0.197  | 0.52  | 66 |
| 3     | Fama-French            |  -11.54  |  0.147  | 0.52  | 66 |
| 4     | IV (2SLS)              |  -15.49  |  0.338  |  —    | 66 |
| 5     | DiD (event-based)      |  -13.05  |  0.365  |  —    | 66 |

**{numref}`tab-regression-robust`: Robustness Checks (see Appendix C tables for detail)**

| Robustness Angle              | Key Result                                        |
|-------------------------------|---------------------------------------------------|
| Alternative returns (Table C.1) | ΔFFR remains negative; significance unchanged    |
| Alternative weights (Table C.2) | Magnitudes stable under value/pure-play tilts    |
| Outliers/robust (Table C.3)     | Winsor/exclusion/Huber keep sign, similar size   |
| Diagnostics ({numref}`tab-diagnostics`)    | No severe multicollinearity; residual tests OK   |

### Appendix G.1: Figure Narratives (Figures 3–9 and Chart K)

- **Chart K (BNPL vs Market with rate-hike shading):** Three aligned panels show BNPL returns (A), market returns (B), and beta-adjusted BNPL residuals (C) with lightly shaded policy periods (COVID, zero bound, hikes).

Co-movement with the market dominates the level plots; in the residual panel the series stays muted even during hikes, reinforcing that most variation is market-driven rather than policy-specific. Zero lines are thick and legends simplified for quick reading.

- **{numref}`fig-observed-fitted` (Observed vs Fitted, full model):** Observed BNPL returns versus fitted values from the full model ({numref}`tab-regression-main`, col 2) cluster around the 45-degree line, yielding R2 ≈ 0.524. Early-period points (blue) and late-period points (orange) overlap tightly.

The largest gaps appear in high-volatility months (COVID rebound, early hikes) where observed returns flare above fitted values in the 5–15% fitted range; outside those tails, fitted and observed move together, showing market and macro controls capture most level variation.

- **{numref}`fig-residuals-ffr` (Residuals vs FFR changes):** Residuals plotted against monthly FFR changes with a LOESS smoother hug the zero line. No slope or curvature emerges; outliers are limited to a few rate-surge months. This pattern supports the adequacy of a linear rate term and is consistent with weak rate significance once controls are included.

- **{numref}`fig-residuals-fitted` (Residuals vs Fitted):** Residuals are symmetric with no funnel shape; variance stays roughly constant across the fitted range. Only the most positive fitted values show modest spread. This aligns with the Breusch–Pagan pass in {numref}`tab-diagnostics` and supports the linear specification and homoskedasticity assumptions used with HC3 SEs.

- **{numref}`fig-qq-plot` (Q–Q Plot):** Residual quantiles track the normal diagonal with only slight tail softness. Jarque–Bera p = 0.429 ({numref}`tab-diagnostics`) means normality cannot be rejected, so t-based inference is reasonable for the full model.

- **{numref}`fig-r2-comparison` (R2 across specifications):** The base FFR-only model explains ~0.02, while the full, Fama–French, and IV variants cluster near ~0.52. The DiD variant drops back near ~0.02. The jump from base to full shows market and macro controls drive explanatory power; rate-only adds very little, and robustness across alternative specs leaves the story unchanged.

- **{numref}`fig-timeline` (Timeline: rates, BNPL vs market, residual):** Panel A plots the Federal Funds Rate; Panel B overlays BNPL and market returns; Panel C shows BNPL returns net of beta-adjusted market exposure. Shading marks COVID (gray), zero bound (blue), and hikes (red).

BNPL closely tracks the market; the residual panel shows limited rate-linked structure, emphasizing that market factors dominate.

In [32]:
from pathlib import Path
import pandas as pd

tables_dir = Path('_build/tables')
tables_dir.mkdir(parents=True, exist_ok=True)

# Table B1
pd.DataFrame([['AFRM Returns', 'Yahoo', 'yfinance', 'AFRM', 'Jan 2021–Present'], ['SEZL Returns', 'Yahoo', 'yfinance', 'SEZL', 'Jul 2019–Present'], ['PYPL Returns', 'Yahoo', 'yfinance', 'PYPL', 'Feb 2015–Present'], ['SPY Returns', 'Yahoo', 'yfinance', 'SPY', 'Jan 1993–Present'], ['Federal Funds', 'FRED', 'fredapi', 'FEDFUNDS', 'Jul 1954–Present'], ['CPI (SA)', 'FRED', 'fredapi', 'CPIAUCSL', 'Jan 1947–Present'], ['Consumer Sent.', 'FRED', 'fredapi', 'UMCSENT', 'Nov 1952–Present'], ['Disp. Income', 'FRED', 'fredapi', 'DSPIC96', 'Jan 1959–Present'], ['FF 3-Factor', 'Ken French', 'pandas', 'Library URL', 'Jul 1926–Present']]).to_csv(tables_dir/'appendix_b1.csv', index=False)

# C1
pd.DataFrame([['Log Returns (Baseline)', '-12.89', '9.99', '0.197', '0.524', '66'], ['Simple Returns', '-12.89', '10.19', '0.206', '0.502', '66'], ['Excess Returns (vs RF proxy)', '-13.54', '10.01', '0.176', '0.525', '66'], ['Market-Adjusted Returns', '-12.89', '9.99', '0.197', '0.345', '66']], columns=['Dependent Variable','Beta (Delta FFR)','SE','p-value','R^2','N']).to_csv(tables_dir/'appendix_c1.csv', index=False)
# C2
pd.DataFrame([['Equal-Weighted (Baseline)', '-12.89', '9.99', '0.197', '0.524', '66'], ['Value-Weighted (proxy)', '-7.74', '5.99', '0.197', '0.622', '66'], ['Pure-Play Tilt (proxy)', '-14.18', '10.99', '0.197', '0.524', '66']], columns=['Weighting Scheme','Beta (Delta FFR)','SE','p-value','R^2','N']).to_csv(tables_dir/'appendix_c2.csv', index=False)
# C3
pd.DataFrame([['Full Sample (Baseline)', '-12.89', '9.99', '0.197', '0.524', '66'], ['Winsorize Returns at 5%', '-11.80', '9.78', '0.227', '0.515', '66'], ['Exclude |R| > 30%', '-13.84', '10.01', '0.167', '0.434', '58'], ['Robust Regression (Huber)', '-12.03', '8.62', 'NaN', 'NaN', '66']], columns=['Specification','Beta (Delta FFR)','SE','p-value','R^2','N']).to_csv(tables_dir/'appendix_c3.csv', index=False)
# D1
pd.DataFrame([['ffr_change', '1.238029'], ['cc_change', '1.084775'], ['di_change', '1.061241'], ['cpi_change', '1.238310'], ['market_return', '1.024770']], columns=['Variable','VIF']).to_csv(tables_dir/'appendix_d1.csv', index=False)
# D2
pd.DataFrame([['1', '0.004488', '0.946587'], ['4', '0.871474', '0.928617'], ['8', '11.249400', '0.187964'], ['12', '15.377028', '0.221463']], columns=['Lag','Q-Statistic','p-value']).to_csv(tables_dir/'appendix_d2.csv', index=False)
# D3
pd.DataFrame([['Breusch-Pagan', 'Chi2=1.67', '0.892348', 'Homoskedastic'], ['White', 'Chi2=13.91', '0.834891', 'Homoskedastic'], ['Goldfeld-Quandt', 'F=0.92', '0.584996', 'Homoskedastic']], columns=['Test','Statistic','p-value','Result']).to_csv(tables_dir/'appendix_d3.csv', index=False)
# E1
pd.DataFrame([['2020-03-31', '2022-02-28', '10.83', '35.93', '0.763', '0.522', '24'], ['2020-04-30', '2022-03-31', '42.40', '122.78', '0.730', '0.500', '24'], ['2020-05-31', '2022-04-30', '10.52', '192.07', '0.956', '0.504', '24'], ['2020-06-30', '2022-05-31', '23.87', '32.08', '0.457', '0.457', '24'], ['2020-07-31', '2022-06-30', '-0.68', '16.44', '0.968', '0.464', '24'], ['2020-08-31', '2022-07-31', '-14.90', '16.19', '0.374', '0.462', '24'], ['2020-09-30', '2022-08-31', '-12.72', '12.45', '0.316', '0.462', '24'], ['2020-10-31', '2022-09-30', '-10.91', '12.28', '0.387', '0.449', '24'], ['2020-11-30', '2022-10-31', '-12.43', '12.19', '0.317', '0.444', '24'], ['2020-12-31', '2022-11-30', '-15.73', '11.58', '0.179', '0.426', '24'], ['2021-01-31', '2022-12-31', '-21.29', '11.74', '0.087', '0.413', '24'], ['2021-02-28', '2023-01-31', '-24.13', '12.79', '0.075', '0.405', '24'], ['2021-03-31', '2023-02-28', '-20.93', '12.16', '0.111', '0.409', '24'], ['2021-04-30', '2023-03-31', '-15.83', '12.14', '0.208', '0.420', '24'], ['2021-05-31', '2023-04-30', '-15.75', '12.08', '0.208', '0.421', '24'], ['2021-06-30', '2023-05-31', '-12.40', '11.28', '0.279', '0.424', '24'], ['2021-07-31', '2023-06-30', '-12.18', '11.18', '0.287', '0.422', '24'], ['2021-08-31', '2023-07-31', '-14.05', '11.01', '0.215', '0.434', '24'], ['2021-09-30', '2023-08-31', '-13.20', '10.77', '0.233', '0.431', '24'], ['2021-10-31', '2023-09-30', '-12.78', '10.61', '0.242', '0.431', '24'], ['2021-11-30', '2023-10-31', '-13.04', '10.61', '0.236', '0.433', '24'], ['2021-12-31', '2023-11-30', '-10.52', '10.37', '0.327', '0.439', '24'], ['2022-01-31', '2023-12-31', '-10.68', '10.41', '0.322', '0.439', '24'], ['2022-02-28', '2024-01-31', '-11.48', '10.38', '0.282', '0.442', '24'], ['2022-03-31', '2024-02-29', '-8.64', '10.14', '0.416', '0.449', '24'], ['2022-04-30', '2024-03-31', '-5.31', '9.97', '0.602', '0.454', '24'], ['2022-05-31', '2024-04-30', '-5.34', '9.92', '0.600', '0.456', '24'], ['2022-06-30', '2024-05-31', '-6.71', '9.74', '0.499', '0.456', '24'], ['2022-07-31', '2024-06-30', '-6.64', '9.62', '0.497', '0.454', '24'], ['2022-08-31', '2024-07-31', '-6.65', '9.56', '0.493', '0.454', '24'], ['2022-09-30', '2024-08-31', '-6.34', '9.47', '0.508', '0.454', '24']], columns=['Window Start','Window End','Beta (Delta FFR)','SE','p-value','R^2','N']).to_csv(tables_dir/'appendix_e1.csv', index=False)
# F1
pd.DataFrame([['BNPL Portfolio', '1.70', '65.99', '0.11', '-42.80', '41.30', '66'], ['Fintech Lenders', '1.21', '47.89', '0.10', '-26.88', '29.80', '66'], ['S&P 500', '1.44', '18.22', '0.49', '-12.35', '12.69', '66'], ['UST 10Y', '0.39', '8.14', '0.39', '-8.07', '10.48', '66'], ['Gold', '1.56', '18.78', '0.67', '-8.03', '10.75', '66'], ['Bitcoin', '8.74', '90.24', '0.72', '-43.84', '58.95', '66']], columns=['Asset Class','Mean (monthly %)','Std Dev (ann. %)','Sharpe (rf=3% ann)','Min (monthly %)','Max (monthly %)','N']).to_csv(tables_dir/'appendix_f1.csv', index=False)
